plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts

```diff
@@ -0,0 +1,366 @@
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          remoteUrl: {
+            title: 'A URL to the repository with the provider',
+            type: 'string',
+          },
+          repoContentsUrl: {
+            title: 'A URL to the root of the repository',
+            type: 'string',
+          },
+          commitHash: {
+            title: 'The git commit hash of the initial commit',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        repoUrl,
+        description,
+        defaultBranch = 'main',
+        gitAuthorName,
+        gitAuthorEmail,
+        gitCommitMessage = 'initial commit',
+        sourcePath,
+      } = ctx.input;
+
+      const { repo, host, owner } = parseRepoUrl(repoUrl, integrations);
+
+      const integrationConfig = integrations.gitea.byHost(host);
+      if (!integrationConfig) {
+        throw new InputError(
+          `No matching integration configuration for host ${host}, please check your integrations config`,
+        );
+      }
+      const { username, password } = integrationConfig.config;
+
+      if (!username || !password) {
+        throw new Error('Credentials for the gitea ${host} required.');
+      }
+
+      // check if the org exists within the gitea server
+      if (owner) {
+        await checkGiteaOrg(integrationConfig.config, { owner });
+      }
+
+      await createGiteaProject(integrationConfig.config, {
+        description,
+        owner: owner,
+        projectName: repo,
+      });
+
+      const auth = {
+        username: username,
+        password: password,
+      };
+      const gitAuthorInfo = {
+        name: gitAuthorName
+          ? gitAuthorName
+          : config.getOptionalString('scaffolder.defaultAuthor.name'),
+        email: gitAuthorEmail
+          ? gitAuthorEmail
+          : config.getOptionalString('scaffolder.defaultAuthor.email'),
+      };
+      // The owner to be used should be either the org name or user authenticated with the gitea server
+      const remoteUrl = `${integrationConfig.config.baseUrl}/${owner}/${repo}.git`;
+      const commitResult = await initRepoAndPush({
+        dir: getRepoSourceDirectory(ctx.workspacePath, sourcePath),
+        remoteUrl,
+        auth,
+        defaultBranch,
+        logger: ctx.logger,
+        commitMessage: generateCommitMessage(config, gitCommitMessage),
+        gitAuthorInfo,
+      });
+
+      // Check if the gitea repo URL is available before to exit
+      const operationTimeLimit = 20000; // 20 seconds
+      const checkDuration = checkDurationLimit(
+        () =>
+          checkAvailabilityGiteaRepository(integrationConfig.config, {
+            owner,
+            repo,
+            defaultBranch,
+            ctx,
+          }),
+        operationTimeLimit,
+      );
+
+      if (!checkDuration) {
+        console.log('Operation exceeded the time limit.');
```

Comment:
Should this really just log, or should it throw an error? Is the output valuable if we don't get the created repo?

Reply:
I removed this code